src/test/java/dev/openfeature/sdk/FlagEvaluationSpecTest.java

```diff
@@ -414,8 +404,9 @@ void broken_provider() {
             text =
                     "In cases of abnormal execution, the `evaluation details` structure's `error message` field **MAY** contain a string containing additional details about the nature of the error.")
     @Test
-    void broken_provider_withDetails() {
-        FeatureProviderTestUtils.setFeatureProvider(new AlwaysBrokenWithDetailsProvider());
+    void broken_provider_withDetails() throws InterruptedException {
+        api.setProvider(new AlwaysBrokenWithDetailsProvider());
+        Thread.sleep(50);
```

Comment:
Why the sleep?

Reply:
I need to investigate this/optimize the test. It was flaky because I either got a flag not found or a provider not ready error, and with `setProviderAndWait` it dit not initialize properly

Reply:
found the culprit and removed it